services/spar/src/Spar/Scim/User.hs

```diff
@@ -556,40 +556,49 @@ updateValidScimUser tokinfo@ScimTokenInfo {stiTeam} uid newValidScimUser =
       oldValidScimUser :: ST.ValidScimUser <-
         validateScimUser "recover-old-value" tokinfo . Scim.value . Scim.thing $ oldScimStoredUser
 
+      -- if the locale of the new valid SCIM user is not set,
+      -- we set it to default value from brig
+      defLocale <- lift BrigAccess.getDefaultUserLocale
+      let newValidScimUser = nvsu {ST._vsuLocale = ST._vsuLocale nvsu <|> Just defLocale}
```

Comment:
I think this should be guaranteed to happen when constructing a `ValidScimUser`.  Just remove the `Maybe` from this record field and follow the type errors.

Reply:
I like the idea, but it is brig responsibility to set the default locale. So we can't require this to be set before the user is stored in brig.